base/compiler/utilities.jl

```diff
@@ -349,6 +369,51 @@ end
 @inline slot_id(s) = isa(s, SlotNumber) ? (s::SlotNumber).id :
     isa(s, Argument) ? (s::Argument).n : (s::TypedSlot).id
 
+######################
+# IncrementalCompact #
+######################
+
+# specifically meant to be used with body1 = compact.result and body2 = compact.new_new_nodes, with nvals == length(compact.used_ssas)
+function find_ssavalue_uses1(body1::Vector{Any}, body2::Vector{Any}, nvals::Int)
+    uses = zeros(Int, nvals)
+
+    function increment_uses(ssa::SSAValue)
+        uses[ssa.id] += 1
+    end
+
+    for line in 1:(length(body1) + length(body2))
+        # index into the right body
+        if line <= length(body1)
+            isassigned(body1, line) || continue
+            e = body1[line]
+        else
+            nline = line - length(body1)
+            isassigned(body2, nline) || continue
+            e = body2[nline]
+        end
+
+        # only look at relevant (in terms of uses) fields
+        if isa(e, QuoteNode)
+            isdefined(e, :value) || continue
+            e = e.value
+        elseif isa(e, GotoIfNot)
+            isdefined(e, :cond) || continue # this may not be necessary
+            e = e.cond
+        elseif isa(e, ReturnNode) || isa(e, PiNode) || isa(e, UpsilonNode)
+            isdefined(e, :val) || continue
+            e = e.val
+        end
+
+        # find uses of SSA values
+        if isa(e, SSAValue)
```

Comment:
move this case into `foreachssa` ?

Reply:
Actually, on second thought, I think this should be covered by just `userefs`?

I.e., if `a = SSAValue(5)`, `:(%5)` is in `userefs(a)`

Reply:
Yes, it probably should be. Not sure if it is right now, but if not, worth fixing.